Add multi-tenant support [LTS] (#6396)
AzureAD/microsoft-authentication-library-for-js: lib/msal-browser/docs/multi-tenant-accounts.md

# Multi-tenant Support in MSAL Browser

> > This is an advanced document regarding the acquisition of tokensacross tenants in `msal-browser`. For basic account information, please review the [Accounts](./accounts.md) document.

Reviewer: ```suggestion
> This is an advanced feature that enables to acquire tokens across tenants in `msal-browser`. For basic account information, please review the [Accounts](./accounts.md) document.
```

## Multi-tenant Accounts

As of v2.39.0, MSAL Browser supports the acquisition of access tokens for accounts across multiple tenants. In order to achieve this, MSAL caches the account information and auth artifacts for a user for each tenant it has authenticated with. 

> This means that with the exception of refresh tokens, which are shared across tenants, the MSAL cache may store multiple accounts, access tokens, and ID tokens for a single user. Each of these account records, in a multi-tenant context, will be referred to as a **"tenant profile"** of the main account.

Reviewer: ```suggestion
With the exception of refresh tokens, which are shared across tenants, the MSAL cache will now store accounts, access tokens, and ID tokens `per tenant` for a single user. Each of these account records, in a multi-tenant context, will be referred to as a **"tenant profile"** of the main account.
```

### Tenant Profiles

> A tenant profile is the record of a particular account in a particular tenant. When multi-tenant support is enabled, a distinct `AccountInfo` object will be cached containing the entire tenant profile for every tenant that the user authenticates with.

Reviewer: ```suggestion
A tenant profile is the record of an account in a particular tenant. When multi-tenant support is enabled, a distinct `AccountInfo` object will be cached containing the entire tenant profile for every tenant that the user authenticates with.
```

> A tenant profile is the record of a particular account in a particular tenant. When multi-tenant support is enabled, a distinct `AccountInfo` object will be cached containing the entire tenant profile for every tenant that the user authenticates with.

Reviewer: What does "containing entire tenant profile" mean? Can we expand that - is it that `tenant_info` or all `tenants`?
